### PR TITLE
Rebrand "Cray Shasta" as "HPE Cray EX".

### PR DESCRIPTION
**Pull request by gbtitus**
We have an official brandname for Shasta now: HPE Cray EX.  Use that in
the appropriate places, and change the internal name from `cray-shasta` to
`hpe-cray-ex` as well.

While here, in the cross-compiling logic in the third-party Makefiles,
use `hpe-cray-ex-unknown-linux-gnu` instead of `...-cle-linux-gnu` for the
configure host type (target type, in Chapel parlance).  This is the same
thing we do for Cray CS targets, and is a better match for EX systems
than what we do for XC ones.

Also while here, specify `CHPL_LIBFABRIC=libfabric` during the builds, so
we use the bundled libfabric.  It's not necessary/desirable to supply a
"system" one from outside the repo any more.

This resolves Cray/chapel-private#1208.

### CODE REVIEW

**Comment by gbtitus** [comment]
@karlonw You may be interested in this.  Note that builds cannot actually be done until this can be cherry-picked back into the release/1.22 branch, however.

File: third-party/libfabric/Makefile [hunk 1]
```diff
@@ -6,12 +6,12 @@ CHPL_MAKE_HOST_TARGET = --target
...
 
 #
-# Cray X* builds are cross-compilations.
+# Cray builds are cross-compilations.
 #
-ifneq (, $(filter cray-x% cray-shasta,$(CHPL_MAKE_TARGET_PLATFORM)))
-CHPL_LIBFABRIC_CFG_OPTIONS += --host=$(shell uname -m)-cle-linux-gnu
-else ifneq ($(CHPL_MAKE_HOST_PLATFORM),$(CHPL_MAKE_TARGET_PLATFORM))
-CHPL_LIBFABRIC_CFG_OPTIONS += --host=$(CHPL_MAKE_TARGET_PLATFORM)-unknown-linux-gnu
+ifneq (, $(filter cray-x% hpe-cray-ex,$(CHPL_MAKE_TARGET_PLATFORM)))
```
Comment (ronawho) [inline third-party/libfabric/Makefile:11]: Move hpe-cray-ex to be with cray-cs (think it's right in all 3p except libfab)

Comment (gbtitus) [reply third-party/libfabric/Makefile]: Oops -- thanks!  Fixed this.

**Review by ronawho — APPROVED** [review]
What's here looks good to me. I did not verify that there's no other outstanding references to shasta in the code base, but assuming you grepped already

**Comment by gbtitus** [comment]
> assuming you grepped [for 'shasta'] already

Yep, did!  Only instances are in `CHANGES.md` and `util/devel/chplspell-dictionary`, and those should stay.

**Pull request merged**